src/parseConfigFile.ts

```diff
@@ -0,0 +1,51 @@
+import * as fs from "fs";
+import { Cell, CellValue, GameState } from "./gamestate";
+
+export function loadGameStateFromFile(filePath: string): GameState {
+  let data = parseConfigFile(filePath);
+  if (!Array.isArray(data)) {
+    throw Error(`Malformed config file ${filePath}`);
+  }
+  let rowsCount = data.length;
+  let gameState = { board: [] };
+  for (let row of data) {
+    gameState.board.push(parseRow(row, rowsCount, filePath));
+  }
+  return gameState;
+}
+
+function parseConfigFile(filePath: string): any {
+  try {
+    return JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  } catch (error) {
+    console.error(error);
+    throw Error(`Unable to parse config file ${filePath}`);
+  }
+}
+
+function parseRow(row: any, rowsCount: number, filePath: string): Array<Cell> {
+  if (!Array.isArray(row)) {
+    throw Error(`Malformed config file ${filePath}`);
+  }
+  if (row.length !== rowsCount) {
+    throw Error(
+      `Malformed config file ${filePath}: all rows should contain ${rowsCount} cells`
+    );
+  }
+  let parsedRow: Array<Cell> = [];
+  for (let cell of row) {
+    let parsedCell = { value: parseCellValue(cell) };
```

Comment:
use const instead of let whenever it is possible

Reply:
why the parseCellValue returns only a value and not a Cell?

Reply:
used `const` when possible and reworked `parseCellValue` function so that it returns a Cell